From 78927c760df4cffd705bc019f0e7ba42ee8a50bc Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Thu, 30 Apr 2015 11:15:37 +0200
Subject: [PATCH 088/117] inspection: Support Fedora 17+ in old inspection code
 (RHBZ#1216305)

Recognize grub2 as a bootloader, and /bin being a symlink to
/usr/bin (ie. UsrMove feature added in Fedora 17+).

Backport to the old inspection code of
commit 5c8a241e5cd119f436a050ddbbf7ea73f83ff8e4 in the C inspect API.
---
 perl/lib/Sys/Guestfs/Lib.pm | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/perl/lib/Sys/Guestfs/Lib.pm b/perl/lib/Sys/Guestfs/Lib.pm
index 11045b9..cd03112 100644
--- a/perl/lib/Sys/Guestfs/Lib.pm
+++ b/perl/lib/Sys/Guestfs/Lib.pm
@@ -459,14 +459,16 @@ sub inspect_partition
 
         # Grub /boot?
         if ($g->is_file ("/grub/menu.lst") ||
-            $g->is_file ("/grub/grub.conf")) {
+            $g->is_file ("/grub/grub.conf") ||
+            $g->is_file ("/grub2/grub.cfg")) {
             $r{content} = "linux-grub";
             _check_grub ($g, \%r);
             goto OUT;
         }
 
         # Linux root?
-        if ($g->is_dir ("/etc") && $g->is_dir ("/bin") &&
+        if ($g->is_dir ("/etc") &&
+            ($g->is_dir ("/bin") || ($g->is_symlink ("/bin") && $g->is_dir ("/usr/bin"))) &&
             $g->is_file ("/etc/fstab")) {
             $r{content} = "linux-root";
             $r{is_root} = 1;
-- 
2.5.0

